src/collision.cpp

```diff
@@ -360,11 +360,16 @@ collisionMoveResult collisionMoveSimple(Environment *env, IGameDef *gamedef,
 #ifndef SERVER
 		ClientEnvironment *c_env = dynamic_cast<ClientEnvironment*>(env);
 		if (c_env != 0) {
-			f32 distance = speed_f->getLength();
```

Comment:
Since 'dtime' wasn't present this looks like it was using a distance that was far too big? Interesting.

Reply:
Yes, that's also why the player (and other entities) lagged up and down while standing on top of each other.